### PR TITLE
Documentation.

### PR DESCRIPTION
**Pull request by Kuyantsev-VP**
PR to demonstrate current progress in working on documentation.

[Link to branch](https://github.com/rvadim/elastalert-k8s-automation/tree/documentation) to see Readme as it is

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,19 +1,137 @@
...
 
-## Building
-To build and upload image to the repository run the following commands:
+The goal of the current project is to simplify configuring ElastAlert notifications about events of some other 
+applications that store its logs in ElasticSearch.
+
+There are two modes for running this application: the local mode and the remote mode.
+* Remote mode is intended for running application in Kubernetes cluster where ElastAlert 
+is started.
+* Local mode allows to run application without deployment in Kubernetes.
```
Comment (rvadim) [inline README.md]: ... For testing purposes.

File: README.md [hunk 2]
```diff
@@ -1,19 +1,137 @@
...
+* Local mode allows to run application without deployment in Kubernetes. 
+
+The application uses two types of configuration files to configure work of ElastAlert:
+* Administrator configuration file with general properties and notification options.
+* User rules configuration files that determine rules for ElastAlert.
+
+The application collects both types of configuration files and generates configuration for ElastAlert.
+
+## Installing and configuring
+Clone elastlert-k8s-automation project:
```
Comment (rvadim) [inline README.md]: Actually for installation we need to just `kubectl apply -f <URL to file with deployment and config>`.
So, in this case we need to prepare docker image building and merge deployment and configmap to one file in `example` directory.
I will prepare docker image building, and you should create example.

File: README.md [hunk 3]
```diff
@@ -1,19 +1,137 @@
...
+*Note: it is assumed that ElastAlert and ElasticSearch are started*
+
+### Administrator configuration file
+The administrator configuration file must contain main options for ElastAlert and must have following structure:
+```yaml
+es_host: # Address of an ElasticSearch cluster
+
+es_port: # Port for es_host
+
+rules_folder: # Directory with ElastAlert rules
```
Comment (rvadim) [inline README.md]: This options not required, due to it calculated. Also, changing this parameter may cause fail to start.

File: README.md [hunk 4]
```diff
@@ -1,19 +1,137 @@
...
+
+run_every: # Frequency of queries to ElasticSearch
+  
+buffer_time: # Size of the query window, stretching backwards from the time each query is run. 
+            #This value is ignored for rules where use_count_query or use_terms_query is set to true.
+
+writeback_index: # ElasticSearch index for ElastAlert logs 
+```
+
+Example of administrator configuration file can be found in
```
Comment (rvadim) [inline README.md]: Link is better.

File: README.md [hunk 5]
```diff
@@ -1,19 +1,137 @@
...
+```
+
+All settings are specified by the “alert_configs” option in the admin config. For each type of alerts that will be used 
+in the system, the administrator must specify a list of configurations (the “configs” option) and a default configuration 
+(the “default” option). The settings for each individual configuration are specified in ElastAlert format, more detailed 
+information can be found in the documentation:
+
+https://elastalert.readthedocs.io/en/latest/ruletypes.html#alerts
+
+Example of general alert settings in the admin config can be found in
```
Comment (rvadim) [inline README.md]: Link

Comment (rvadim) [reply README.md]: But inserting example inline here is better then link for this.

File: README.md [hunk 6]
```diff
@@ -1,19 +1,137 @@
...
+User rules should be set in the configmap named “elastalert-rules”. Each such configmap must contain a set of named rules.
+
+Rules are created in the format of ElastAlert user rules and can contain any valid ElastAlert options.
+
+In addition, for alerts there is a special option {alert}_id (where {alert} is the type of alert supported by ElastAlert, 
+for example, email). {alert}_id determines which system settings specified by the administrator will be used for this 
+alert. If {alert}_id is not specified or refers to a nonexistent system settings, the default settings will be used.
+Examples of user rules can be found in
+
+/tests/fixtures/renderer_test_user_rules.yaml
```
Comment (rvadim) [inline README.md]: Insert example inline

**Review by rvadim — CHANGES_REQUESTED** [review]

File: README.md [hunk 7]
```diff
@@ -1,19 +1,191 @@
...
+The application uses two types of configuration files to configure work of ElastAlert:
+* Administrator configuration file with general properties and notification options.
+* User rules configuration files that determine rules for ElastAlert.
+
+The application collects both types of configuration files and generates configuration for ElastAlert.
+
+## Installing and configuring
+* To run application remotely, you have to install docker image in Kubernetes:<br/>
+``
+$ kubectl apply -f <URL to file with deployment and config>
```
Comment (rvadim) [inline README.md]: https://github.com/rvadim/elastalert-k8s-automation/blob/master/deploy/deploy.yaml

Comment (rvadim) [reply README.md]: But need to merge configmap to deploy.yaml

**Review by rvadim — CHANGES_REQUESTED** [review]

**Review by rvadim — APPROVED** [review]
LGTM

File: README.md (hunk 7 above)
Comment (dkondyrev) [reply README.md]: Changed

**Review by dkondyrev — APPROVED** [review]

**Pull request merged**